I18n.md

```diff
@@ -0,0 +1,124 @@
+# I18n
+
+UI uses [react-i18n](https://react.i18next.com/) to make the components internationalizable. This is based on [i18next](https://www.i18next.com/)
+
+The App defines the strategy how to get the translations.
+
+### Namespaces
+
+Each packages has their own namespace:
+
+| Package        | Namespace     |
+| ------------- 	|:-------------:|
+| components      | tui-components |
+| forms      			| tui-forms      |
+
+### Creating a i18n components
+
+Each components who has to be internationalizable are exported by default with translate from 'react-i18next' with his namespace.
+
+```javascript
+import React from 'react';
+import { translate } from 'react-i18next';
+
+const HelloWorld = ({t}) => (
+ <div>{`Hi ${t('HELLO', { defaultValue: 'Hello world' })}`}</div>
+);
+
+// Export component to test it with a fake t method
+export { HelloWorld };
+export default translate(NAMESPACE)(HelloWorld);
+```
+
+### Translate
+
+In the framework, at each time that the `t` method is called, a defaultValue is given to test it.
+
+```javascript
+t('SUFFIX_COMPONENT_KEY', { defaultValue: 'translated'});
+```
+
+The key has to be prefixed by the main current component to translate.
+e.g:
+ LIST_DISPLAY
+ HEADERBAR_GO_PORTAL
+
+## Use i18n with UI in your App
+
+### Create an instance of i18n
+`yarn add i18next`
+
+```javascript
+import i18n from 'i18next';
+
+i18n.init({
+	debug: false,
+	wait: true, // globally set to wait for loaded translations in translate hoc
+});
+
+export default i18n;
+```
+
+The UI framework have to be nested into a Provider to allow to translate the labels. The provider is responsible to pass the i18next instance down to all the translate hocs using react context.
+
+### Use the HOC TranslateWrapper
+
+```javascript
+import { TranslateWrapper } from 'react-talend-component';
+import i18n from ...
+
+import MyComponentToTranslate from ...
+
+const MyComponentTranslated = TranslateWrapper(MyComponentToTranslate, { i18n });
```

Comment:
Question: TranslateWrapper is for ngReact only right ?

Reply:
nope It is a HOC to integrate the i18n provider to the translatable components.